Source/Blazorise/BarDropdown.razor.cs

```diff
@@ -29,15 +29,24 @@ protected override void BuildClasses( ClassBuilder builder )
             base.BuildClasses( builder );
         }
 
-        protected override void OnInitialized()
+        protected override Task OnInitializedAsync()
         {
             // link to the parent component
             ParentBarItem?.Hook( this );
 
-            if ( parentStore.Mode != BarMode.VerticalSmall )
-                Visible = Open;
+            return base.OnInitializedAsync();
+        }
+
+        public override Task SetParametersAsync( ParameterView parameters )
+        {
+            // This is needed for the two-way binding to work properly.
+            // Otherwise the internal value would not be set in the right order.
+            if ( parameters.TryGetValue<bool>( nameof( Visible ), out var newVisible ) )
+            {
+                store.Visible = newVisible;
+            }
 
-            base.OnInitialized();
+            return base.SetParametersAsync( parameters );
```

Comment:
@stsrki Thank you for this! I couldn't figure this part out!

Will this no longer be needed once https://github.com/dotnet/aspnetcore/pull/19642 is addressed?

Reply:
No, component activator solves another problem. This is more of how Blazor works with parameters.

Actually, I had a similar problem with `TextEdit `a year ago so I just figured it might work.